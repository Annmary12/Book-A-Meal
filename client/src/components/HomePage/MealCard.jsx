import React, { Component } from 'react';
import PropTypes from 'prop-types';

const MealCard = (props) =>  (
  <div className="card">
    <div className="card-img">
      <img src={props.image} />
    </div>
    <div className="card-body">
      <div className="card-title">{props.title}</div>
      <div className="card-description">
        {props.description}
      </div>
    </div>
    <div className="card-footer">
      <span>
        <b>Price</b>: &#8358;{props.price}
      </span>
      <div className="card-options">
        <a className="order-btn" href="login.html">Order</a>
      </div>
    </div>
  </div>
)

MealCard.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
}

export default MealCare