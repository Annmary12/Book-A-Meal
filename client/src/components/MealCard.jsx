import React, { Component } from 'react';

class MenuPeep extends Component{
  render () {
    return (
      <div className="card">
        <div className="card-img">
          <img src={this.props.image} />
        </div>
        <div className="card-body">
          <div className="card-title">{this.props.title}</div>
          <div className="card-description">
            {this.props.description}
          </div>
        </div>
        <div className="card-footer">
          <span>
            <b>Price</b>: &#8358;{this.props.price}
          </span>
          <div className="card-options">
            <a className="order-btn" href="login.html">Order</a>
          </div>
        </div>
      </div>
    )
  }
}

export default MenuPeep;