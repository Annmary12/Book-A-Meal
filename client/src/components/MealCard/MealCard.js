import React, { PureComponent, Fragment } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import OrderModal from '../Orders/OrderModal/OrderModal.js';
import localizeNum from '../../utils/localizeNum.js';
import ConfirmDeleteModal from '../../components/ConfirmDeleteModal/ConfirmDeleteModal.js';
import EditMealModal from './EditMealModal/EditMealModal.js';
import cancelIcon from '../../../assets/img/cancel-icon.svg';
import pencilIcon from '../../../assets/img/pencil-icon.svg';
import styles from './meal-card.scss';

/**
 * @class MealCard
 */
class MealCard extends PureComponent {
  static propTypes = {
    meal: PropTypes.shape({
      img: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired
    }).isRequired,
    action: PropTypes.string,
    className: PropTypes.string,
    deleteMeal: PropTypes.func
  };

  static defaultProps = {
    action: '',
    className: '',
    deleteMeal: () => {}
  };

  state = {
    orderModalOpen: false,
    deleteModalOpen: false,
    editModalOpen: false
  }

  openOrderModal = () => {
    this.setState({ orderModalOpen: true });
  }

  closeOrderModal = () => {
    this.setState({ orderModalOpen: false });
  }

  openDeleteModal = () => {
    this.setState({ deleteModalOpen: true });
  }

  closeDeleteModal = () => {
    this.setState({ deleteModalOpen: false });
  }

  openEditModal = () => {
    this.setState({ editModalOpen: true });
  }

  closeEditModal = () => {
    this.setState({ editModalOpen: false });
  }

  handleDelete = () => {
    const { meal } = this.props;

    this.props.deleteMeal(meal);
  }

  /**
   * @returns {JSX} React JSX
   */
  render() {
    const {
      meal,
      action,
      className: addedClassName
    } = this.props;

    return (
      <div className={classnames(styles.card, addedClassName)}>
        <div className={styles.image}>
          <img src={meal.img} alt="meal" />
        </div>
        <div className={styles.body}>
          <div className={styles.title}>
            {meal.title}
          </div>
          <div className={styles.description}>
            {meal.description}
          </div>
        </div>
        <div className={styles.footer}>
          <span>
            <b>Price</b>: &#8358;{localizeNum(meal.price)}
          </span>
          <div className={styles.actions}>
            {
              action === 'order' &&
              <button
                className={styles.orderBtn}
                onClick={this.openOrderModal}
              >
                  Order
              </button>
            }
            {
              action === 'edit' &&
              <Fragment>
                <button
                  className={styles.iconBtn}
                  onClick={this.openEditModal}
                >
                  <img
                    src={pencilIcon}
                    width="18"
                    alt="edit"
                    className={styles.icon}
                  />
                </button>
                <button
                  className={styles.iconBtn}
                  onClick={this.openDeleteModal}
                >
                  <img
                    src={cancelIcon}
                    width="18"
                    alt="delete"
                    className={styles.icon}
                  />
                </button>
                <ConfirmDeleteModal
                  isOpen={this.state.deleteModalOpen}
                  confirmText="Are you sure you want to delete meal?"
                  handleClose={this.closeDeleteModal}
                  handleDelete={this.handleDelete}
                />
                <EditMealModal
                  isOpen={this.state.editModalOpen}
                  handleClose={this.closeEditModal}
                  meal={meal}
                />
              </Fragment>
            }
          </div>
        </div>
        {
          action === 'order' &&
          <OrderModal
            isOpen={this.state.orderModalOpen}
            handleClose={this.closeOrderModal}
            meal={meal}
          />
        }

      </div>
    );
  }
}


export default MealCard;
