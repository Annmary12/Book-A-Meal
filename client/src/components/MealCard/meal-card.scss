@keyframes fadeIn{
  from{
    opacity: 0;
    transform: translate(40px);
  }

  to{
    opacity: 1;
  }
}

.card {
  background-color: white;
  border-radius: 5px;
  margin: auto;
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;

  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  // transition: all 0.3s cubic-bezier(.25,.8,.25,1);
  animation: fadeIn .8s ease-in;

  &:hover{
    box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
  }

  @media only screen and (max-width: 600px) {
    width: 85%;
  }


  @media only screen and (min-width: 600px) {
    width: 90%;
  }

  @media only screen and (min-width: 768px) {
    width: 90%;
  }

  @media only screen and (min-width: 992px) {
    width: 85%;
  }

}

.image{
  height: 218px;

  img{
    border-radius: 5px 5px 0 0;
    height: 100%;
    width: 100%;
  }
}

.body{
  padding: 5px;

  @media only screen and (min-width: 600px) {
		padding: 14px;
  }

  @media only screen and (min-width: 768px) {
    padding: 14px;
  }
}

.title{
  font-weight: bold;
  margin-bottom: 5px;
  font-size: 16px;
  height: 1em;
  line-height: 1em;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  // @media only screen and (max-width: 600px){
  //   font-size: 19px;
  // }
}

.description{
  color: #3f3e3e;
  font-size: 13px;
  height: 4em;
  line-height: 1em;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #3f3e3e;
}

.footer{
  display: flex;
  padding: 10px;
  border-top: 1px solid #d0d0d0;
}

.actions{
  margin-left: auto;
}

.orderBtn{
	background-color: #FF5722;
  border: 2px solid #b93c15;
  color: white;
  border-radius: 5px;
  padding-left: 9px;
  padding-right: 9px;
  padding-top: 2px;
  padding-bottom: 2px;
  font-size: 16px;
  text-decoration: none;

  &:hover{
    background-color: #b93c15;
  }
}
