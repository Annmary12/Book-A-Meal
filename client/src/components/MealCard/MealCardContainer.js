import { connect } from 'react-redux';
import { deleteMeal } from '../../actions/meals.js';
import MealCard from './MealCard.js';

export default connect(null, { deleteMeal })(MealCard);
