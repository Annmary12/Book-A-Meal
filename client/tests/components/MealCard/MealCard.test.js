import React from 'react';
import MealCard from '../../../src/components/MealCard/MealCard';

const meal = {
  img: 'picture.jpg',
  title: 'ewedu',
  description: 'adsasdfs',
  price: 3900
};

describe('MealCard component', () => {
  const cardWrapper = shallow(
    <MealCard
      meal={meal}
    />
  );

  it('should match snapshot if action prop is edit', () => {
    const wrapper = shallow(
      <MealCard
        meal={meal}
        action="edit"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if action prop is order', () => {
    const wrapper = shallow(
      <MealCard
        meal={meal}
        action="order"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('openOrderModal method', () => {
    it('should set orderModalOpen state to true when called', () => {
      cardWrapper.instance().openOrderModal();

      const state = cardWrapper.instance().state;

      expect(state.orderModalOpen).toBeTruthy();
    });
  });

  describe('closeOrderModal method', () => {
    it('should set orderModalOpen state to false when called', () => {
      cardWrapper.instance().closeOrderModal();

      const state = cardWrapper.instance().state;

      expect(state.orderModalOpen).toBeFalsy();
    });
  });

  describe('openDeleteModal method', () => {
    it('should set deleteModalOpen state to true when called', () => {
      cardWrapper.instance().openDeleteModal();

      const state = cardWrapper.instance().state;

      expect(state.deleteModalOpen).toBeTruthy();
    });
  });

  describe('closeDeleteModal method', () => {
    it('should set orderModalOpen state to false when called', () => {
      cardWrapper.instance().closeDeleteModal();

      const state = cardWrapper.instance().state;

      expect(state.deleteModalOpen).toBeFalsy();
    });
  });

  describe('openEditModal method', () => {
    it('should set editModalOpen state to true when called', () => {
      cardWrapper.instance().openEditModal();

      const state = cardWrapper.instance().state;

      expect(state.editModalOpen).toBeTruthy();
    });
  });

  describe('closeEditModal method', () => {
    it('should set editModalOpen state to false when called', () => {
      cardWrapper.instance().closeEditModal();

      const state = cardWrapper.instance().state;

      expect(state.editModalOpen).toBeFalsy();
    });
  });

  describe('closeEditModal method', () => {
    it('should set editModalOpen state to false when called', () => {
      cardWrapper.instance().closeEditModal();

      const state = cardWrapper.instance().state;

      expect(state.editModalOpen).toBeFalsy();
    });
  });

  describe('handleDelete method', () => {
    it('should call deleteMeal prop function with meal', () => {
      const props = {
        deleteMeal: jest.fn(),
        meal
      };

      const wrapper = shallow(
        <MealCard
          {...props}
        />
      );

      wrapper.instance().handleDelete();

      expect(props.deleteMeal).toBeCalledWith(props.meal)
    });
  })
});