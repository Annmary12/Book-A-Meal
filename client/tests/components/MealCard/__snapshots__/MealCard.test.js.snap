// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCard component should match snapshot if action prop is edit 1`] = `
<div
  className="card"
>
  <div
    className="image"
  >
    <img
      alt="meal"
      src="picture.jpg"
    />
  </div>
  <div
    className="body"
  >
    <div
      className="title"
    >
      ewedu
    </div>
    <div
      className="description"
    >
      adsasdfs
    </div>
  </div>
  <div
    className="footer"
  >
    <span>
      <b>
        Price
      </b>
      : ₦
      3,900.00
    </span>
    <div
      className="actions"
    >
      <button
        className="iconBtn"
        onClick={[Function]}
      >
        <img
          alt="edit"
          className="icon"
          src="test-file-stub"
          width="18"
        />
      </button>
      <button
        className="iconBtn"
        onClick={[Function]}
      >
        <img
          alt="delete"
          className="icon"
          src="test-file-stub"
          width="18"
        />
      </button>
      <ConfirmDeleteModal
        confirmText="Are you sure you want to delete meal?"
        handleClose={[Function]}
        handleDelete={[Function]}
        isOpen={false}
      />
      <EditMealModal
        handleClose={[Function]}
        isOpen={false}
        meal={
          Object {
            "description": "adsasdfs",
            "img": "picture.jpg",
            "price": 3900,
            "title": "ewedu",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`MealCard component should match snapshot if action prop is order 1`] = `
<div
  className="card"
>
  <div
    className="image"
  >
    <img
      alt="meal"
      src="picture.jpg"
    />
  </div>
  <div
    className="body"
  >
    <div
      className="title"
    >
      ewedu
    </div>
    <div
      className="description"
    >
      adsasdfs
    </div>
  </div>
  <div
    className="footer"
  >
    <span>
      <b>
        Price
      </b>
      : ₦
      3,900.00
    </span>
    <div
      className="actions"
    >
      <button
        className="orderBtn"
        onClick={[Function]}
      >
        Order
      </button>
    </div>
  </div>
  <OrderModal
    handleClose={[Function]}
    isOpen={false}
    meal={
      Object {
        "description": "adsasdfs",
        "img": "picture.jpg",
        "price": 3900,
        "title": "ewedu",
      }
    }
  />
</div>
`;
